packages/wallet/src/api.ts

```diff
@@ -12,36 +11,111 @@ import {
-  signTxTheCIP30Way: (
-    tx: MarloweTxCBORHex
-  ) => TE.TaskEither<Error, HexTransactionWitnessSet>;
-  getChangeAddress: T.Task<AddressBech32>;
-  getUsedAddresses: T.Task<AddressBech32[]>;
-  getCollaterals: T.Task<TxOutRef[]>;
-  getUTxOs: T.Task<TxOutRef[]>;
-  getCIP30Network: T.Task<CIP30Network>;
-  getTokens: TE.TaskEither<Error, Token[]>;
-  getLovelaces: TE.TaskEither<Error, bigint>;
+  // DISCUSSION: Should this function receive a timeout and checkinterval?
+  // DISCUSSION: Should this be a part of the base API or an extended functionality on top of
+  //              `getUTxOs` similar to `getAddressesAndCollaterals`
+  /**
+   * Waits for a transaction to be confirmed.
+   * @experimental
+   * @param txHash the transaction hash
+   * @returns true if the transaction is confirmed, false otherwise
+   */
+  waitConfirmation(txHash: string): Promise<boolean>;
+  // DISCUSSION: should we create different Nominal types for different CBOR representations and mix and match accordingly?
+  //             maybe the WalletAPI should be generic on the CBOR representation, and it is necesary to make sure that the
+  //             generation and consumption of those CBOR are compatible.
+  // TODO: See if we can we improve the Error? CIP30 talks about DataSignError and TxSignError
+  //  https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#datasignerror
+  //  https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#txsignerror
+  /**
+   * Ask the wallet to sign a transaction
+   * @param tx A CBOR encoded transaction
+   * @returns The signed transaction or throws
+   * @throws `Error` if there was a problem signing the transaction
+   *
+   * @experimental
+   *
+   * @remarks
+   * CIP30 expects a CBOR Transaction, which consists of <txBody + witness>. The
+   * Marlowe runtime default flow returns just the txBody, to get what CIP30 expects
+   * you need to pass an Accept header when creating a contract or applying inputs.
+   * The header is `Accept "application/vendor.iog.marlowe-runtime.contract-tx-json"`
+   * and it is automatically included in the @marlowe.io/runtime-* packages.
+   */
+  signTxTheCIP30Way(tx: MarloweTxCBORHex): Promise<HexTransactionWitnessSet>;
+  /**
+   * Returns an address to be used as Change Address when balancing transactions.
+   */
+  getChangeAddress(): Promise<AddressBech32>;
+  /**
+   * Some wallets have a single address and some wallets derive multiple addresses from a single root private key.
+   * This method helps to keep track of the used derived addresses.
+   */
+  getUsedAddresses(): Promise<AddressBech32[]>;
+  /**
+   * A list of UTXO to use as collaterals
+   * @see {@link https://docs.cardano.org/plutus/collateral-mechanism/}
+   */
+  getCollaterals(): Promise<TxOutRef[]>;
+  /**
+   * @returns The list of UTXOs available to the wallet.
+   */
+  getUTxOs(): Promise<TxOutRef[]>;
+  // DISCUSSION: should we keep CIP30 as part of the name of the api?
```

Comment:
I believe, we don't care it is about CIP30... We care that it is Browser vs NodeJS, that's why I have removed  CIP30 from the naming etc...